repo/open.go

```diff
@@ -183,51 +186,125 @@ func openDirect(ctx context.Context, configFile string, lc *LocalConfig, passwor
 	return r, nil
 }
 
-// openWithConfig opens the repository with a given configuration, avoiding the need for a config file.
-// nolint:funlen,gocyclo
-func openWithConfig(ctx context.Context, st blob.Storage, lc *LocalConfig, password string, options *Options, caching *content.CachingOptions, configFile string) (DirectRepository, error) {
-	caching = caching.CloneOrDefault()
-
-	// Read repo blobs, potentially from cache.
-	fb, rb, err := readAndCacheRepositoryBlobs(ctx, st, caching.CacheDirectory, lc.FormatBlobCacheDuration)
+// nolint:gocritic
+func readAndCacheRepoConfig(
+	ctx context.Context,
+	st blob.Storage,
+	password string,
+	cacheOpts *content.CachingOptions,
+	validDuration time.Duration,
+) (
```

Comment:
can this return a proper structure for readability?

Reply:
done in [4dabb7d](https://github.com/kopia/kopia/pull/1758/commits/4dabb7d600c13aa983325619ac79ce3fcd271cdc)